docs-source/src/architecture.rst

```diff
@@ -39,12 +65,15 @@ Grace time
 **********
 A job can take some time to check out the license from the license server after it is submitted to Slurm.
 Thus, each license has a ``grace time`` period that is used to indicate how long a booking will be retained.
-After the ``grace time`` expires, the booking is deleted. This means that the license was checked out from the
-license server and doesn't need a booking anymore.
+After the ``grace time`` expires, the booking is deleted. If the ``grace time`` expires the job is assumed to have failed
+and deemed not to consume the requested licenses, hence the booking is deleted to allow for other jobs start. Note that
```

Comment:
```suggestion
After the ``grace time`` expires, the booking is deleted. If the ``grace time`` expires the job is assumed to have failed
and the requested licenses are returned to the available license pool. In this case, the booking is deleted to allow for other jobs to start. Note that
```

Reply:
I think this section need some more work to point out the details of the behavior. I will try to rephrase it further.